internal/sync/lifecycle_test.go

```diff
@@ -146,15 +147,39 @@ func TestLifecycleOnce(t *testing.T) {
 				StartAction{ExpectedState: Running},
 				ConcurrentAction{
 					Actions: []LifecycleAction{
-						StopAction{ExpectedState: Stopped, Wait: 20 * time.Millisecond},
+						StopAction{ExpectedState: Stopped, Wait: 15 * time.Millisecond},
```

Comment:
++ this introduces only 5 ms of  breathing room before the state moves to Stopped.  Since this is GetState and not GetExactState there should be too much flakiness here per se, but it's something to keep in mind